DEVELOPMENT_GUIDE.md

```diff
@@ -0,0 +1,221 @@
+# Development Guide
+
+This documentation is intended for developers to develop the schema, sandbox and proxies. It may be used by developers working within the NHS Digital organisation and outside contributors.
+
+## Table of Contents
+
+- [Development Guide](#development-guide)
+  - [Table of Contents](#table-of-contents)
+  - [Development](#development)
+    - [Requirements](#requirements)
+    - [Install](#install)
+      - [Updating hooks](#updating-hooks)
+    - [Environment Variables](#environment-variables)
+    - [Make commands](#make-commands)
+    - [Testing](#testing)
+    - [VS Code Plugins](#vs-code-plugins)
+    - [Emacs Plugins](#emacs-plugins)
+    - [Speccy](#speccy)
+    - [Caveats](#caveats)
+      - [Swagger UI](#swagger-ui)
+      - [Apigee Portal](#apigee-portal)
+    - [Platform setup](#platform-setup)
+    - [Detailed folder walk through](#detailed-folder-walk-through)
+      - [`/.github`:](#github)
+      - [`/azure`:](#azure)
+      - [`/proxies`:](#proxies)
+      - [`/sandbox`:](#sandbox)
+        - [Development](#development-1)
+      - [`/scripts`:](#scripts)
+      - [`/specification`:](#specification)
+      - [`/tests`:](#tests)
+      - [`Makefile`:](#makefile)
+      - [`ecs-proxies-containers.yml ` and `ecs-proxies-deploy.yml`:](#ecs-proxies-containersyml--and-ecs-proxies-deployyml)
+      - [`manifest_template.yml`:](#manifest_templateyml)
+      - [Package management:](#package-management)
+  - [Releasing a new schema version](#releasing-a-new-schema-version)
+
+## Development
+
+### Requirements
+
+-   GNU make
+-   nodejs + npm/yarn
+-   Python 3.8 + [poetry](https://github.com/python-poetry/poetry)
+-   Java 8+
+
+### Install
+
+```
+$ make install
+```
+
+#### Updating hooks
+
+You can install some pre-commit hooks to ensure you can't commit invalid spec changes by accident. These are also run
+in CI, but it's useful to run them locally too.
+
+```
+$ make install
+```
+
+### Environment Variables
+
+Various scripts and commands rely on environment variables being set. These are documented with the commands.
+
+:bulb: Consider using [direnv](https://direnv.net/) to manage your environment variables during development and maintaining your own `.envrc` file - the values of these variables will be specific to you and/or sensitive.
+
+### Make commands
+
+There are `make` commands that alias some of this functionality:
+
+-   `lint` -- Lints the spec and code
+-   `publish` -- Outputs the specification as a **single file** into the `build/` directory
+-   `serve` -- Serves a preview of the specification in human-readable format
+
+### Testing
+
+Each API and team is unique. We encourage you to use a `test/` folder in the root of the project, and use whatever testing frameworks or apps your team feels comfortable with. It is important that the URL your test points to be configurable. We have included some stubs in the Makefile for running tests.
```

Comment:
This section feels like boilerplate from the APIM template. Is there concrete guidance on testing for our setup we can give?